### DISCUSSION TITLE
Using avalanche for personal data

### DISCUSSION THREAD

**Discussion by nxa190008** [ContinualAI/avalanche#1057] [Q&A]
I have a signal dataset available based on which I have to build a model that can classify any unknown data. I guess I have to do task incremental continual learning. Can I input short-time fourier transform data to Avalanche? If yes, should I use the multi-head scenario?

**Comment by AndreaCossu** [comment]
The use of task-incremental scenarios and multi-headed models really depends on your specific problem. The task can be tackled under different conditions. First, which type of task are you dealing with (sequence classification/regression, sequence modeling, other)? Avalanche accepts any kind of tensor in input as dataset. My suggestion is to check out the [`dataset_benchmark`](https://avalanche-api.continualai.org/en/latest/generated/avalanche.benchmarks.generators.dataset_benchmark.html?highlight=tensor_benchmark) and the [`AvalancheDataset`](https://avalanche-api.continualai.org/en/latest/generated/avalanche.benchmarks.utils.avalanche_dataset.AvalancheDataset.html?highlight=avalanche%20dataset). You can create different datasets (one for each experience) with the latter and put them together in a benchmark with the former.

**Reply by nxa190008** [reply]
Hello

Thanks for your reply. Actually, I am totally new to python, let alone avalanche. I am trying to summarize what I have and what I want to do here:
1. I have .mat files of stft images from 3 classes, for each class I have 3 different datasets (I guess different datasets are called 'experiences'). I just loaded them using loadmat and used the  function "avalanchetensordataset" to create the avalanchedataset. Is that correct?
2. Which function should I call to create the benchmark dataset? How do I create the training and test streams?
3. I want to use the 'simplecnn' model and 'lwf' strategy for now. How should I proceed?
 Can you please summarize how to do these steps? Or provide a link to a full example? I searched the whole website but couldn't understand much.
